core/unit_test/cuda/TestCuda_InterOp_Graph.cpp

```diff
@@ -148,4 +148,15 @@ TEST_F(TEST_CATEGORY_FIXTURE(GraphInterOp), instantiation_flags) {
 #endif
 }
 
+// Build a Kokkos::Graph from an existing cudaGraph_t.
+TEST(GraphInterOp, construct_from_native) {
+  cudaGraph_t native_graph = nullptr;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGraphCreate(&native_graph, 0));
+
+  auto graph = Kokkos::Experimental::create_graph_from_native(Kokkos::Cuda{},
+                                                              native_graph);
+
+  ASSERT_EQ(native_graph, graph.native_graph());
+}
```

Comment:
What about other tests that actually execute the graph?

Reply:
Done. Note that this PR does not allow us to build `Kokkos` nodes from nodes that are already in the native graph.